application/__init__.py

```diff
@@ -26,15 +26,22 @@
 from werkzeug.exceptions import HTTPException
 from werkzeug.serving import WSGIRequestHandler
 
+logger = logging.getLogger(__name__)
+
 try:
-    with open('log_config.yaml', 'r') as config_file:
+    with open('log_config.yaml') as config_file:
         config = yaml.safe_load(config_file.read())
     logging.config.dictConfig(config)
-except Exception as ex:
-    print("Logging setup failed with exception = ", ex)
-
-logger = logging.getLogger(__name__)
-logger.warning(f"Logging is set up with config={config}")
+except Exception:
+    # Fallback to a basic configuration
+    logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s')
+    logger.exception("Logging setup failed")
+else:
+    # Initialize the logger after dictConfig succeeds
+    # As well, to ensure that logging configuration succeeded regardless
+    # of a successful dictConfig execution, log a warning message
+    # to see whether the configuration was actually successful
+    logger.warning("Logging is set up with config=%s", config)
```

Comment:
it might be more clear to express this as
```suggestion
    logger.warning("Logging setup is completed with config=%s", config)
```
which also matches the above error message